app.js

```diff
@@ -0,0 +1,68 @@
+const express = require("express");
+const cors = require("cors");
+// const axios = require("axios");
+const body  = require("body-parser");
+const mySchool = require('./School')
+
+const port = 3000;
+
+const app = express();
+app.use(cors());
+app.use(body.urlencoded({extended: false}));
+
+// let mySchool = new School();
+
+const getAllClasses = (req, res) => {
+    console.log("these are the classes" + mySchool.classes)
+    res.json({
+        allClasses: mySchool.classes
+        
+    })
+}
+
+const checkIfClassExists = (req, res, next) =>  {
+        
+    if (req.class) {
+        //here logic for if input != req.body.class then add the class
+        res.status(200).json({//chain it to send back a json and let it know it's all good
+        message: "Success your class has been added!",
+        data: req.class
+
+        })
+    } else {
+        res.status(400).json({
+            message: "That class already exsists, add another"
+        })
+        }
+    next();
+}
+
+// const addNewClass = (req, res) => {
+//do you just need/want a fucntion??
+// }
+
+const addNewClass = (req, res) => {
+    console.log(req.params);
+}
+
+app.get("/addClass/:className/:teacher", (req, res) => {
+    console.log(req.params);
+    className = req.params.className;
+    teacher = req.params.teacher;
+
+    newClass = [{"className": className, "teacher":teacher}]
+    res.json(newClass);
+        
+    //am I doing this right to get a new class inside of the class object
```

Comment:
1. You are adding new data to the backend, not simply getting information. You this would be a POST request.
2. You need to declare your variables
3. In your School.js file you have a method that describes how the action of adding a new class happens. You need to use that function, and pass the arguments it requires - which is the name of the class and a teacher.